<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"> 
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"> 
    
    <link rel="stylesheet" href="../../../fonts/academicons-1.8.6/css/academicons.min.css"/>
    <link rel="icon" type="image/png" sizes="32x32" href="../../../logo/bodhi.png"> 
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    
    
    <title>UCSCXenaTools: R API for UCSC Xena Hubs - Shixiang Wang</title>
    
     
    <meta property="og:title" content="UCSCXenaTools: R API for UCSC Xena Hubs - Shixiang Wang | 王诗翔">
    

    
      
    

    

    
    


<link href='//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css' rel='stylesheet' type='text/css' />



    <link rel="stylesheet" href="../../../css/style.css" />
    <link rel="stylesheet" href="../../../css/mystyle.css" /> 
    <link rel="stylesheet" href="../../../css/fonts.css" />
    
<link rel="stylesheet" href="../../../css/custom.css" />

  </head>

  
  <body class="en">
    <header class="masthead">
      

<h1><a href="../../../"><img src="https://avatars1.githubusercontent.com/u/25057508?s=460&amp;v=4" alt="Shixiang Wang" /></a></h1>



      <nav class="menu">
        <input id="menu-check" type="checkbox" />
        <label id="menu-label" for="menu-check" class="unselectable">
          <span class="icon close-icon">✕</span>
          <span class="icon open-icon">☰</span>
          <span class="text">Menu</span>
        </label>
        <ul>
        
        
        <li><a href="../../../">Home</a></li>
        
        <li><a href="../../../en/about/">About</a></li>
        
        <li><a href="../../../en/post/">Blog</a></li>
        
        <li><a href="../../../en/tools/">Software</a></li>
        
        <li><a href="../../../en/list/">List</a></li>
        
        <li><a href="../../../en/cv-en/shixiang">CV</a></li>
        
        <li><a href="../../../cn/">中文</a></li>
        
        

<li class="menu-extra"></li>



<li><a href="https://github.com/ShixiangWang/home/edit/master/content/en/tools/ucscxenatools-api.utf8.md" target="_blank">Edit Me</a></li>


<li><a href="../../../en/index.xml" type="application/rss+xml" title="RSS feed">Subscribe</a></li>

<li><a href="http://creativecommons.org/licenses/by-nc-sa/4.0/" title="Attribution-NonCommercial-ShareAlike 4.0 International">License</a></li>


        </ul>
      </nav>
    </header>

    <article class="main">
      <header class="title">
      

      </header>


<div style="text-align: center; width: 150px; float:right;">
<a href = "../ucscxenatools-api.pdf" class = "fa fa-file-pdf-o" title = "PDF version "> PDF </a>
</div>



<p>This vignette gives users the summary information of API functions provided by <strong>UCSCXenaTools</strong> for UCSC Xena.</p>

<p>Before using API, user should know some concepts about Xena elements. Following description is copied from <a href="https://github.com/ucscXena/xenaPython/blob/master/xenaPython/__init__.py">xenaPython <code>__init__.py</code></a>.</p>

<blockquote>
<p>Data rows are associated with &ldquo;sample&rdquo; IDs.</p>

<p>Sample IDs are unique within a &ldquo;<strong>cohort</strong>&rdquo;.
s
A &ldquo;<strong>dataset</strong>&rdquo; is a particular assay of a cohort, e.g. gene expression.</p>

<p>Datasets have associated metadata, specifying their data type and cohort.</p>

<p>There are three primary data types: <strong>dense matrix</strong> (samples by probes), <strong>sparse</strong> (sample, position, variant), and <strong>segmented</strong> (sample, position, value).</p>

<p>Dense matrices can be genotypic or phenotypic. Phenotypic matrices have associated <strong>field metadata</strong> (descriptive names, codes, etc.).
Genotypic matricies may have an associated <strong>probeMap</strong>, which maps probes to genomic locations. If a matrix has hugo probeMap, the probes themselves are gene names. Otherwise, a probeMap is used to map a gene location to a set of probes.</p>
</blockquote>

<h1 id="new-features">New features</h1>

<p>A new series of functions (mostly) starting with <code>fetch_</code> have been introduced to help fetch a small amount of data by Xena APIs.</p>

<p>Now three are available:</p>

<ul>
<li><code>fetch_dense_values()</code>: fetches values from a dense matrix.</li>
<li><code>fetch_dataset_samples()</code>: fetches samples from a dataset</li>
<li><code>fetch_dataset_identifiers()</code>: fetches identifies from a dataset.</li>
</ul>

<p>They have similar arguments and all the details can be viewed by running <code>?fetch</code> in R console after <code>library(UCSCXenaTools)</code>.</p>

<h1 id="api-categories">API categories</h1>

<p>API functions can be divided into two classes: <strong>lower API functions</strong> and <strong>higher API functions</strong>.
They have following difference:</p>

<ul>
<li>The main difference between them is that the target of higher API functions is <code>XenaHub</code> object, which is
a S4 class built in R. While the targets of lower API functions are Xena hub urls, cohort names or
dataset names with character format. The <code>XenaHub</code> object can provide more uniform operation methods and can be
used to download corresponding datasets quickly and easily
(detail see <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro/">another vignette</a>).</li>
<li>Lower API functions are not registered in package
<a href="https://github.com/ShixiangWang/UCSCXenaTools/blob/master/NAMESPACE">NAMESPACE</a>, so user may not access them
after <code>library(UCSCXenaTools)</code>, user need to use <code>UCSCXenaTools:::fun_name</code> instead.</li>
<li>Lower API functions have no help pages, so user cannot find any description about them in R, which means
you cannot use <code>?fun_name</code> to get help. However, API report part in this vignette shows all avaiable API functions and
their short description.</li>
<li>Higher API functions are built on lower API functions, they return more meaningful and easy results for operation. Most of lower API functions return nested lists as results, user need to tidy them before using them in next step.</li>
</ul>

<h2 id="lower-api-functions">Lower API functions</h2>

<p>Lower API functions also have 2 classes:</p>

<ul>
<li><p>one is generated from
<a href="https://github.com/ShixiangWang/UCSCXenaTools/tree/master/inst/queries"><code>.xq</code> files</a>,
function names all start with <code>.p_</code>. All <code>.xq</code> files are copied from <a href="https://github.com/ucscXena/xenaPython">xenaPython</a> package, which is official Python API for Xena.
These functions are dynamicly created when <strong>UCSCXenaTools</strong> loaded. Their names are given as following:</p>

<pre><code>#&gt;  [1] &quot;.p_all_cohorts&quot;                    
#&gt;  [2] &quot;.p_all_datasets&quot;                   
#&gt;  [3] &quot;.p_all_datasets_n&quot;                 
#&gt;  [4] &quot;.p_all_field_metadata&quot;             
#&gt;  [5] &quot;.p_cohort_samples&quot;                 
#&gt;  [6] &quot;.p_cohort_summary&quot;                 
#&gt;  [7] &quot;.p_dataset_fetch&quot;                  
#&gt;  [8] &quot;.p_dataset_field&quot;                  
#&gt;  [9] &quot;.p_dataset_field_examples&quot;         
#&gt; [10] &quot;.p_dataset_field_n&quot;                
#&gt; [11] &quot;.p_dataset_gene_probe_avg&quot;         
#&gt; [12] &quot;.p_dataset_gene_probes_values&quot;     
#&gt; [13] &quot;.p_dataset_list&quot;                   
#&gt; [14] &quot;.p_dataset_metadata&quot;               
#&gt; [15] &quot;.p_dataset_probe_signature&quot;        
#&gt; [16] &quot;.p_dataset_probe_values&quot;           
#&gt; [17] &quot;.p_dataset_samples&quot;                
#&gt; [18] &quot;.p_dataset_samples_ndense_matrix&quot;  
#&gt; [19] &quot;.p_datasets_null_rows&quot;             
#&gt; [20] &quot;.p_feature_list&quot;                   
#&gt; [21] &quot;.p_field_codes&quot;                    
#&gt; [22] &quot;.p_field_metadata&quot;                 
#&gt; [23] &quot;.p_gene_transcripts&quot;               
#&gt; [24] &quot;.p_match_fields&quot;                   
#&gt; [25] &quot;.p_probe_count&quot;                    
#&gt; [26] &quot;.p_probemap_list&quot;                  
#&gt; [27] &quot;.p_ref_gene_exons&quot;                 
#&gt; [28] &quot;.p_ref_gene_position&quot;              
#&gt; [29] &quot;.p_ref_gene_range&quot;                 
#&gt; [30] &quot;.p_segment_data_examples&quot;          
#&gt; [31] &quot;.p_segmented_data_range&quot;           
#&gt; [32] &quot;.p_sparse_data&quot;                    
#&gt; [33] &quot;.p_sparse_data_examples&quot;           
#&gt; [34] &quot;.p_sparse_data_match_field&quot;        
#&gt; [35] &quot;.p_sparse_data_match_field_slow&quot;   
#&gt; [36] &quot;.p_sparse_data_match_partial_field&quot;
#&gt; [37] &quot;.p_sparse_data_range&quot;              
#&gt; [38] &quot;.p_transcript_expression&quot;
</code></pre></li>

<li><p>the other one is created in package. The function names all start with <code>.</code>, are given as following:</p>

<pre><code>#&gt; [1] &quot;.host_cohorts&quot;         
#&gt; [2] &quot;.cohort_datasets&quot;      
#&gt; [3] &quot;.cohort_datasets_count&quot;
#&gt; [4] &quot;.cohort_samples_each&quot;  
#&gt; [5] &quot;.cohort_samples_any&quot;   
#&gt; [6] &quot;.cohort_samples_all&quot;   
#&gt; [7] &quot;.dataset_samples_each&quot; 
#&gt; [8] &quot;.dataset_samples_any&quot;  
#&gt; [9] &quot;.dataset_samples_all&quot;
</code></pre></li>
</ul>

<p>I don&rsquo;t know how to write these query sentence for Xena Hubs. So here I want to
say thanks to authors of <a href="https://github.com/ucscXena/xenaPython"><strong>xenaPython</strong></a>
and <a href="https://github.com/mtmorgan/XenaR"><strong>xenaR</strong></a> packages.</p>

<h1 id="api-report">API report</h1>

<p>\includegraphics{ucscxenatools-api_files/figure-latex/unnamed-chunk-4-1}</p>

<p>Of note, I don&rsquo;t know test all functions generated from <code>.xq</code> files, most of them works.
Sometimes functions return you errors or <code>list()</code> may caused by invaild format or bad network,
you should try more times. If you make sure there are problems/errors in query procedure,
you can check corresponding query variables:</p>

<pre><code>#&gt;  [1] &quot;.xq_all_cohorts&quot;                    
#&gt;  [2] &quot;.xq_all_datasets&quot;                   
#&gt;  [3] &quot;.xq_all_datasets_n&quot;                 
#&gt;  [4] &quot;.xq_all_field_metadata&quot;             
#&gt;  [5] &quot;.xq_cohort_samples&quot;                 
#&gt;  [6] &quot;.xq_cohort_summary&quot;                 
#&gt;  [7] &quot;.xq_dataset_fetch&quot;                  
#&gt;  [8] &quot;.xq_dataset_field&quot;                  
#&gt;  [9] &quot;.xq_dataset_field_examples&quot;         
#&gt; [10] &quot;.xq_dataset_field_n&quot;                
#&gt; [11] &quot;.xq_dataset_gene_probe_avg&quot;         
#&gt; [12] &quot;.xq_dataset_gene_probes_values&quot;     
#&gt; [13] &quot;.xq_dataset_list&quot;                   
#&gt; [14] &quot;.xq_dataset_metadata&quot;               
#&gt; [15] &quot;.xq_dataset_probe_signature&quot;        
#&gt; [16] &quot;.xq_dataset_probe_values&quot;           
#&gt; [17] &quot;.xq_dataset_samples&quot;                
#&gt; [18] &quot;.xq_dataset_samples_ndense_matrix&quot;  
#&gt; [19] &quot;.xq_datasets_null_rows&quot;             
#&gt; [20] &quot;.xq_feature_list&quot;                   
#&gt; [21] &quot;.xq_field_codes&quot;                    
#&gt; [22] &quot;.xq_field_metadata&quot;                 
#&gt; [23] &quot;.xq_gene_transcripts&quot;               
#&gt; [24] &quot;.xq_match_fields&quot;                   
#&gt; [25] &quot;.xq_probe_count&quot;                    
#&gt; [26] &quot;.xq_probemap_list&quot;                  
#&gt; [27] &quot;.xq_ref_gene_exons&quot;                 
#&gt; [28] &quot;.xq_ref_gene_position&quot;              
#&gt; [29] &quot;.xq_ref_gene_range&quot;                 
#&gt; [30] &quot;.xq_segment_data_examples&quot;          
#&gt; [31] &quot;.xq_segmented_data_range&quot;           
#&gt; [32] &quot;.xq_sparse_data&quot;                    
#&gt; [33] &quot;.xq_sparse_data_examples&quot;           
#&gt; [34] &quot;.xq_sparse_data_match_field&quot;        
#&gt; [35] &quot;.xq_sparse_data_match_field_slow&quot;   
#&gt; [36] &quot;.xq_sparse_data_match_partial_field&quot;
#&gt; [37] &quot;.xq_sparse_data_range&quot;              
#&gt; [38] &quot;.xq_transcript_expression&quot;
</code></pre>

<p>For example, you&rsquo;d like to check <code>.p_all_cohorts</code> function, you can take a look
at <code>.xq_all_cohorts</code> object.</p>

<pre><code class="language-r">.xq_all_cohorts
#&gt; [1] &quot;;allCohorts\n(fn [exclude]\n\t(map :cohort\n\t  (query\n\t\t{:select [[#sql/call [:distinct #sql/call [:ifnull :cohort \&quot;(unassigned)\&quot;]] :cohort]]\n\t\t :from [:dataset]\n\t\t :where [:not [:in :type exclude]]})))\n&quot;
</code></pre>

<p><code>cat</code> it may give you more easy-to-read format.</p>

<pre><code class="language-r">cat(.xq_all_cohorts)
#&gt; ;allCohorts
#&gt; (fn [exclude]
#&gt; 	(map :cohort
#&gt; 	  (query
#&gt; 		{:select [[#sql/call [:distinct #sql/call [:ifnull :cohort &quot;(unassigned)&quot;]] :cohort]]
#&gt; 		 :from [:dataset]
#&gt; 		 :where [:not [:in :type exclude]]})))
</code></pre>

<h1 id="use-cases">Use cases</h1>

<p>Several use cases are modified from <a href="https://github.com/ucscXena/xenaPython">README</a> of <strong>xenaPython</strong> package.</p>

<p>Load package firstly.</p>

<pre><code class="language-r">library(UCSCXenaTools)
</code></pre>

<p>You can find out host id and dataset id from <a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>, a more recommened way is use <code>XenaData</code> in <strong>UCSCXenaTools</strong>.</p>

<pre><code class="language-r">head(XenaData)[, 1:5]
#&gt; # A tibble: 6 x 5
#&gt;   XenaHosts XenaHostNames XenaCohorts
#&gt;   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;      
#&gt; 1 https://~ publicHub     Acute lymp~
#&gt; 2 https://~ publicHub     Acute lymp~
#&gt; 3 https://~ publicHub     Acute lymp~
#&gt; 4 https://~ publicHub     Breast Can~
#&gt; 5 https://~ publicHub     Breast Can~
#&gt; 6 https://~ publicHub     Breast Can~
#&gt; # ... with 2 more variables:
#&gt; #   XenaDatasets &lt;chr&gt;, SampleCount &lt;chr&gt;
</code></pre>

<p>The host id is stored at <code>XenaHosts</code> column, and dataset id is stored at <code>XenaDatasets</code> column.</p>

<p><strong>Of note, when you want to query single sample or gene with function starts with <code>.p_</code>, you must transform id of sample or gene into a <code>list</code></strong></p>

<h2 id="query-four-samples-and-three-identifers-expression">Query four samples and three identifers expression</h2>

<pre><code class="language-r">hub = &quot;https://toil.xenahubs.net&quot;
dataset = &quot;tcga_RSEM_gene_tpm&quot;
samples = c(&quot;TCGA-02-0047-01&quot;, &quot;TCGA-02-0055-01&quot;, 
    &quot;TCGA-02-2483-01&quot;, &quot;TCGA-02-2485-01&quot;)
probes = c(&quot;ENSG00000282740.1&quot;, &quot;ENSG00000000005.5&quot;, 
    &quot;ENSG00000000419.12&quot;)
.p_dataset_probe_values(hub, dataset, samples, 
    probes)
#&gt; [[1]]
#&gt;   strand  chromend chromstart chrom
#&gt; 1      -  16750589   16739938  chr1
#&gt; 2      -  50958555   50934867 chr20
#&gt; 3      + 100599885  100584802  chrX
#&gt; 
#&gt; [[2]]
#&gt;        [,1]   [,2]   [,3]   [,4]
#&gt; [1,] -9.966 -2.826 -9.966 -9.966
#&gt; [2,] -3.171  4.165 -5.574 -3.171
#&gt; [3,]  4.675  6.025  5.826  5.177
</code></pre>

<p>Query one probe. As metioned above, one must transform id of proble or sample int a <code>list</code> when he wants to query only one sample/probe.</p>

<p><strong>Bad query</strong>:</p>

<pre><code class="language-r">.p_dataset_probe_values(hub, dataset, samples, 
    &quot;ENSG00000282740.1&quot;)
#&gt; [[1]]
#&gt; list()
#&gt; 
#&gt; [[2]]
#&gt;       [,1] [,2] [,3] [,4]
#&gt;  [1,]  NaN  NaN  NaN  NaN
#&gt;  [2,]  NaN  NaN  NaN  NaN
#&gt;  [3,]  NaN  NaN  NaN  NaN
#&gt;  [4,]  NaN  NaN  NaN  NaN
#&gt;  [5,]  NaN  NaN  NaN  NaN
#&gt;  [6,]  NaN  NaN  NaN  NaN
#&gt;  [7,]  NaN  NaN  NaN  NaN
#&gt;  [8,]  NaN  NaN  NaN  NaN
#&gt;  [9,]  NaN  NaN  NaN  NaN
#&gt; [10,]  NaN  NaN  NaN  NaN
#&gt; [11,]  NaN  NaN  NaN  NaN
#&gt; [12,]  NaN  NaN  NaN  NaN
#&gt; [13,]  NaN  NaN  NaN  NaN
#&gt; [14,]  NaN  NaN  NaN  NaN
#&gt; [15,]  NaN  NaN  NaN  NaN
#&gt; [16,]  NaN  NaN  NaN  NaN
#&gt; [17,]  NaN  NaN  NaN  NaN
</code></pre>

<p><strong>Good query</strong>:</p>

<pre><code class="language-r">.p_dataset_probe_values(hub, dataset, samples, 
    as.list(&quot;ENSG00000282740.1&quot;))
#&gt; [[1]]
#&gt;   strand chromend chromstart chrom
#&gt; 1      - 16750589   16739938  chr1
#&gt; 
#&gt; [[2]]
#&gt;        [,1]   [,2]   [,3]   [,4]
#&gt; [1,] -9.966 -2.826 -9.966 -9.966
</code></pre>

<h2 id="query-four-samples-and-three-genes-expression-when-the-dataset-you-want-to-query-has-a-identifier-to-gene-mapping">Query four samples and three genes expression, when the dataset you want to query has a identifier-to-gene mapping</h2>

<blockquote>
<p>identifier-to-gene mapping (i.e. xena probeMap)</p>
</blockquote>

<pre><code class="language-r">genes = c(&quot;TP53&quot;, &quot;RB1&quot;, &quot;PIK3CA&quot;)
.p_dataset_gene_probe_avg(hub, dataset, samples, 
    genes)
#&gt;     gene                      position
#&gt; 1   TP53    -, 7687550, 7661779, chr17
#&gt; 2    RB1  +, 48481986, 48303751, chr13
#&gt; 3 PIK3CA +, 179240093, 179148114, chr3
#&gt;                       scores
#&gt; 1 5.799, 4.428, 6.515, 6.309
#&gt; 2 5.867, 4.700, 4.810, 4.920
#&gt; 3 3.547, 3.377, 2.789, 2.951
</code></pre>

<h2 id="if-the-dataset-does-not-have-id-to-gene-mapping-but-the-dataset-used-gene-names-as-its-identifier">If the dataset does not have id-to-gene mapping, but the dataset used gene names as its identifier</h2>

<p>In this situation, you can query gene expression like two ways above will not work.</p>

<pre><code class="language-r">hub = &quot;https://toil.xenahubs.net&quot;
dataset = &quot;tcga_RSEM_Hugo_norm_count&quot;
samples = c(&quot;TCGA-02-0047-01&quot;, &quot;TCGA-02-0055-01&quot;, 
    &quot;TCGA-02-2483-01&quot;, &quot;TCGA-02-2485-01&quot;)
probes = c(&quot;TP53&quot;, &quot;RB1&quot;, &quot;PIK3CA&quot;)

.p_dataset_probe_values(hub, dataset, samples, 
    probes)
#&gt; [[1]]
#&gt;   strand  chromend chromstart chrom
#&gt; 1      +  48481986   48303751 chr13
#&gt; 2      -   7687550    7661779 chr17
#&gt; 3      + 179240093  179148114  chr3
#&gt; 
#&gt; [[2]]
#&gt;       [,1]  [,2]  [,3]  [,4]
#&gt; [1,] 11.63 10.68 12.65 12.15
#&gt; [2,] 12.04 10.93 11.59 11.41
#&gt; [3,] 10.67 10.90 10.71 10.12
</code></pre>

<h2 id="find-out-the-samples-in-a-dataset">Find out the samples in a dataset</h2>

<pre><code class="language-r">hub = &quot;https://tcga.xenahubs.net&quot;
dataset = &quot;TCGA.BLCA.sampleMap/HiSeqV2&quot;
.p_dataset_samples(hub, dataset, 10)
#&gt;  [1] &quot;TCGA-BT-A20R-11&quot; &quot;TCGA-DK-AA6S-01&quot;
#&gt;  [3] &quot;TCGA-DK-A6B2-01&quot; &quot;TCGA-GU-A763-01&quot;
#&gt;  [5] &quot;TCGA-XF-A9T4-01&quot; &quot;TCGA-FD-A5C1-01&quot;
#&gt;  [7] &quot;TCGA-GU-A42Q-01&quot; &quot;TCGA-DK-A3IL-01&quot;
#&gt;  [9] &quot;TCGA-XF-AAMH-01&quot; &quot;TCGA-FT-A61P-01&quot;
# obtain all samples
.p_dataset_samples(hub, dataset, NULL) %&gt;% head()
#&gt; [1] &quot;TCGA-BT-A20R-11&quot; &quot;TCGA-DK-AA6S-01&quot;
#&gt; [3] &quot;TCGA-DK-A6B2-01&quot; &quot;TCGA-GU-A763-01&quot;
#&gt; [5] &quot;TCGA-XF-A9T4-01&quot; &quot;TCGA-FD-A5C1-01&quot;
</code></pre>

<p>Higher API function <code>samples()</code> has more features. It can be used to do set operation for samples in a host.</p>

<pre><code class="language-r">xe = XenaHub(cohorts = &quot;Cancer Cell Line Encyclopedia (CCLE)&quot;)
# samples in each dataset, first host
x = samples(xe, by = &quot;datasets&quot;, how = &quot;each&quot;)[[1]]
lengths(x)  # data sets in ccle cohort on first (only) host
</code></pre>

<h2 id="find-out-the-identifiers-in-a-dataset">Find out the identifiers in a dataset</h2>

<pre><code class="language-r">hub = &quot;https://tcga.xenahubs.net&quot;
dataset = &quot;TCGA.BLCA.sampleMap/HiSeqV2&quot;
.p_dataset_field(hub, dataset) %&gt;% head()
#&gt; [1] &quot;?|100130426&quot; &quot;?|100133144&quot; &quot;?|100134869&quot;
#&gt; [4] &quot;?|10357&quot;     &quot;?|10431&quot;     &quot;?|136542&quot;
</code></pre>

<h2 id="find-out-the-number-of-idnetifiers-in-a-dataset">Find out the number of idnetifiers in a dataset</h2>

<pre><code class="language-r">hub = &quot;https://tcga.xenahubs.net&quot;
dataset = &quot;TCGA.BLCA.sampleMap/HiSeqV2&quot;
.p_dataset_field_n(hub, dataset)
#&gt; [1] 20531
</code></pre>

<h1 id="license">LICENSE</h1>

<p>GPL-3</p>

<p>Please note, code from <strong>XenaR</strong> package under Apache 2.0 license.</p>


  <footer>
  
<nav class="post-nav">
  <span class="nav-prev"></span>
  <span class="nav-next"><a href="../../../en/tools/ucscxenatools-api.knit/">UCSCXenaTools: R API for UCSC Xena Hubs</a> &rarr;</span>
</nav>
<script type="text/javascript">
document.addEventListener('keyup', function(e) {
  if (e.target.nodeName.toUpperCase() != 'BODY') return;
  var url = false;
  if (e.which == 37) {  
    
  } else if (e.which == 39) {  
    
    url = '\/en\/tools\/ucscxenatools-api.knit\/';
    
  }
  if (url) window.location = url;
});
</script>



<section class="comments">
  <div id="disqus_thread"></div>
  <script src="../../../js/disqusloader.min.js"></script>
  <script>
  var disqus_config = function () {
  
    this.page.url = "https:\/\/shixiangwang.github.io\/home\/" + location.pathname;
  
  };
  (function() {
    var inIFrame = function() {
      var iframe = true;
      try { iframe = window.self !== window.top; } catch (e) {}
      return iframe;
    };
    if (inIFrame()) return;
    var disqus_js = '//home-xjdzylqrzp.disqus.com/embed.js';
    
    if (location.hash.match(/^#comment/)) {
      var d = document, s = d.createElement('script');
      s.src = disqus_js; s.async = true;
      s.setAttribute('data-timestamp', +new Date());
      (d.head || d.body).appendChild(s);
    } else {
      disqusLoader('#disqus_thread', {
        scriptUrl: disqus_js, laziness: 0, disqusConfig: disqus_config
      });
    }
  })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>




<script async src="../../../js/fix-toc.js"></script>

<script async src="../../../js/center-img.js"></script>

<script async src="../../../js/right-quote.js"></script>

<script async src="../../../js/no-highlight.js"></script>

<script async src="../../../js/fix-footnote.js"></script>

<script async src="../../../js/math-code.js"></script>

<script async src="../../../js/external-link.js"></script>

<script async src="../../../js/alt-title.js"></script>

<script async src="../../../js/header-link.js"></script>


<script async src="//cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"></script>

  



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>




  
  
  

  <div class="copyright"><a href="mailto:w_shixiang@163.com"><i class='far fa-envelope fa-1x'></i></a> &middot; <a href="https://github.com/ShixiangWang"><i class='fab fa-github fa-1x'></i></a> &middot; <a href="https://stackoverflow.com/users/7662327/shixiang-wang"><i class='fab fa-stack-overflow fa-1x'></i></a> &middot; <a href="https://scholar.google.com/citations?user=FvNp0NkAAAAJ&amp;hl=en"><i class='ai ai-google-scholar ai-1x'></i></a> &middot; <a href="https://orcid.org/0000-0002-9275-3557"><i class='ai ai-orcid ai-1x'></i></a> &middot; <a href="https://www.researchgate.net/profile/Wang_Shixiang4"><i class='ai ai-researchgate ai-1x'></i></a> <br> Powered by <a href="https://gohugo.io">Hugo</a> and <a href="https://bookdown.org/yihui/blogdown/">Blogdown</a> &copy; <a href="../../../">Shixiang Wang</a> 2017 - 2019 </div>
  
  

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=56h9es09xn7&amp;m=6&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=1" async="async"></script>
  </footer>
  </article>
  
  </body>
</html>

