<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"> 
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"> 
    
    <link rel="stylesheet" href="../../../fonts/academicons-1.8.6/css/academicons.min.css"/>
    <link rel="icon" type="image/png" sizes="32x32" href="../../../logo/bodhi.png"> 
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    
    
    <title>一个函数获取 TCGA/CCLE 单基因分子数据 - 王诗翔</title>
    
     
    <meta property="og:title" content="一个函数获取 TCGA/CCLE 单基因分子数据 - Shixiang Wang | 王诗翔">
    

    
      
    

    

    
    


<link href='//cdn.bootcss.com/highlight.js/9.12.0/styles/Xcode.min.css' rel='stylesheet' type='text/css' />



    <link rel="stylesheet" href="../../../css/style.css" />
    <link rel="stylesheet" href="../../../css/mystyle.css" /> 
    <link rel="stylesheet" href="../../../css/fonts.css" />
    
<script async src="../../../js/load-typekit.js"></script>


<link rel="stylesheet" href="../../../css/custom.css" />

  </head>
  
  <body class="cn">
    <header class="masthead">
      

<h1><a href="../../../logo/ShixiangWang.png"><img src="../../../logo/ShixiangWang.png" alt="Shixiang Wang" /></a></h1>
<p align="right" style="margin-top:-50px;"><small>><i>上士闻道<br>勤而行之</i></small></p>




      <nav class="menu">
        <input id="menu-check" type="checkbox" />
        <label id="menu-label" for="menu-check" class="unselectable">
          <span class="icon close-icon">✕</span>
          <span class="icon open-icon">☰</span>
          <span class="text">Menu</span>
        </label>
        <ul>
        
        
        <li><a href="../../../">首页</a></li>
        
        <li><a href="../../../cn/about/">关于</a></li>
        
        <li><a href="../../../cn/post/">博客</a></li>
        
        <li><a href="../../../cn/read/">读书</a></li>
        
        <li><a href="../../../cn/writing">写作</a></li>
        
        <li><a href="../../../cn/research/">研究</a></li>
        
        <li><a href="../../../cn/tools/">工具</a></li>
        
        <li><a href="../../../cn/cv-cn/shixiang">简历</a></li>
        
        <li><a href="../../../logo/qrcode.jpg">公众号</a></li>
        
        <li><a href="https://www.zhihu.com/people/shixiangwang">知乎</a></li>
        
        <li><a href="../../../cn/mark">推荐阅读</a></li>
        
        <li><a href="../../../en/">English</a></li>
        
        <li><a href="../../../categories/">分类</a></li>
        
        <li><a href="../../../tags/">标签</a></li>
        
        

<li class="menu-extra"></li>






<li><a href="https://github.com/ShixiangWang/home/edit/master/content/cn/post/2020-09-26-access-gene-level-cancer-data-from-ucsc-xena.Rmd" target="_blank">编辑</a></li>


<li><a href="../../../cn/index.xml" type="application/rss+xml" title="RSS feed">订阅</a></li>

<li><a href="http://creativecommons.org/licenses/by-nc-sa/4.0/" title="Attribution-NonCommercial-ShareAlike 4.0 International">版权</a></li>


        </ul>
      </nav>
    </header>

    <article class="main">
      <header class="title">
        

<h1>一个函数获取 TCGA/CCLE 单基因分子数据</h1>



<h3>王诗翔 &middot 
2020-09-26</h3> 


<p style="text-align:right;">
  分类:
  
    <a href="../../../categories/r">r</a> &nbsp
  
  <br>
  标签:
  
    <a href="../../../tags/r">r</a> &nbsp
  
    <a href="../../../tags/%E5%8D%95%E5%9F%BA%E5%9B%A0%E5%88%86%E6%9E%90">单基因分析</a> &nbsp
  
    <a href="../../../tags/xena">Xena</a> &nbsp
  
    <a href="../../../tags/cancer">cancer</a> &nbsp
  
    <a href="../../../tags/genomics">genomics</a> &nbsp
  
</p>



   
  


      </header>





<blockquote>
<p>这篇文章很久之前就想写了，但一直因为一些事情拖延了。</p>
</blockquote>
<p>在开发 UCSCXenaShiny 的基础上，我将其中支持的 UCSCXena TCGA/CCLE 单基因数据下载函数进行了整理，构建了一个单一的入口。这样即使用户无需加载 Shiny，也能够简单自在的下载 癌症单基因数据了。</p>
<blockquote>
<p>这里单独说的 TCGA 不太全面，实际包含了 TCGA TARGET GTEx 3 个数据库，它们是个体水平的数据。而 CCLE 是细胞水平数据。</p>
</blockquote>
<div id="下载安装包" class="section level2">
<h2>下载安装包</h2>
<p>在国内我们统一推荐下载 Gitee 上的包：</p>
<pre class="r"><code>remotes::install_git(&quot;https://gitee.com/XenaShiny/UCSCXenaShiny&quot;)</code></pre>
<p>如果你已经安装 CRAN 上的 UCSCXenaShiny，也需要进行上面的操作，否则无法使用最新的函数。</p>
</div>
<div id="参数说明" class="section level2">
<h2>参数说明</h2>
<p>函数就一个 <code>query_value()</code>。</p>
<p>简单看看有哪些参数：</p>
<pre class="r"><code>library(UCSCXenaShiny)
args(query_value)
#&gt; function (identifier, data_type = c(&quot;gene&quot;, &quot;transcript&quot;, &quot;protein&quot;, 
#&gt;     &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;), database = c(&quot;toil&quot;, &quot;ccle&quot;)) 
#&gt; NULL</code></pre>
<p>非常简单哈，只有 3 个：</p>
<ul>
<li>第 1 个是基因名/蛋白名等。</li>
<li>第 2 个是数据的类型，包括基因表达（gene），转录本表达（transcript）、突变（mutation）、拷贝数变异（cnv）和甲基化（methylation），默认是基因表达。</li>
<li>第 3 个是数据库，包括 toil（包括上面提到的 TCGA 等几个个体水平数据）和 ccle。</li>
</ul>
</div>
<div id="使用" class="section level2">
<h2>使用</h2>
<p>了解函数参数后，使用就根据自己所需就行了。如果还不懂，可以不断试错。</p>
<p>我们以 TP53 基因为例下载一些数据看看。</p>
<pre class="r"><code>gene_expr &lt;- query_value(&quot;TP53&quot;)
#&gt; Running mode: client
#&gt; =========================================================================================
#&gt; UCSCXenaTools version 1.3.3
#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools
#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html
#&gt; 
#&gt; If you use it in published research, please cite:
#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data
#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.
#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627
#&gt; =========================================================================================
#&gt;                               --Enjoy it--
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. ProbeMap is found.
#&gt; Running mode: client
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;TcgaTargetGtex_rsem_gene_tpm&quot;) %&gt;% XenaBrowse()</code></pre>
<p>这个返回结果的结构：</p>
<pre class="r"><code>str(gene_expr)
#&gt; List of 2
#&gt;  $ expression: Named num [1:19131] 4.79 5.89 5.52 4.43 2.38 ...
#&gt;   ..- attr(*, &quot;names&quot;)= chr [1:19131] &quot;GTEX-S4Q7-0003-SM-3NM8M&quot; &quot;TCGA-19-1787-01&quot; &quot;TCGA-S9-A7J2-01&quot; &quot;GTEX-QV31-1626-SM-2S1QC&quot; ...
#&gt;  $ unit      : chr &quot;log2(tpm+0.001)&quot;</code></pre>
<p>可以查看部分数据：</p>
<pre class="r"><code>gene_expr$expression[1:5]
#&gt; GTEX-S4Q7-0003-SM-3NM8M         TCGA-19-1787-01         TCGA-S9-A7J2-01 
#&gt;                    4.79                    5.89                    5.52 
#&gt; GTEX-QV31-1626-SM-2S1QC         TCGA-G3-A3CH-11 
#&gt;                    4.43                    2.38</code></pre>
<p>有了这个数据就可以结合病人的各种表型去做分析啦。</p>
<p>我们再看下它的 CNV 和 突变情况。</p>
<pre class="r"><code># CNV
gene_cnv &lt;- query_value(&quot;TP53&quot;, data_type = &quot;cnv&quot;)
#&gt; Running mode: client
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. No probeMap found or error happened, use old way...
#&gt; Running mode: client
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;TCGA.PANCAN.sampleMap/Gistic2_CopyNumber_Gistic2_all_thresholded.by_genes&quot;) %&gt;% XenaBrowse()
gene_cnv$data[1:5]
#&gt; TCGA-A5-A0GI-01 TCGA-S9-A7J2-01 TCGA-06-0150-01 TCGA-AR-A1AH-01 TCGA-EK-A2RE-01 
#&gt;               0               0               0              -1               0

# 突变
gene_mut &lt;- query_value(&quot;TP53&quot;, data_type = &quot;mutation&quot;)
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;mc3.v0.2.8.PUBLIC.nonsilentGene.xena&quot;) %&gt;% XenaBrowse()
#&gt; Running mode: client
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. ProbeMap is found.
#&gt; Running mode: client
gene_mut[1:5]
#&gt; TCGA-02-0003-01 TCGA-02-0033-01 TCGA-02-0047-01 TCGA-02-0055-01 TCGA-02-2470-01 
#&gt;               1               1               0               1               0</code></pre>
<p>简单利用 tidyverse 包，我们可以将几种分子数据整合到一起：</p>
<pre class="r"><code>library(tidyverse)
#&gt; ── Attaching packages ──────────────────────────────────────────────────────────── tidyverse 1.3.0 ──
#&gt; ✓ ggplot2 3.3.2     ✓ purrr   0.3.4
#&gt; ✓ tibble  3.0.3     ✓ dplyr   1.0.2
#&gt; ✓ tidyr   1.1.2     ✓ stringr 1.4.0
#&gt; ✓ readr   1.3.1     ✓ forcats 0.5.0
#&gt; ── Conflicts ─────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
#&gt; x dplyr::filter() masks stats::filter()
#&gt; x dplyr::lag()    masks stats::lag()

expr &lt;- dplyr::tibble(
  sample = names(gene_expr$expression),
  expr = as.numeric(gene_expr$expression)
)

cnv &lt;- dplyr::tibble(
  sample = names(gene_cnv$data),
  cnv = as.numeric(gene_cnv$data)
)

mut &lt;- dplyr::tibble(
  sample = names(gene_mut),
  mut = as.numeric(gene_mut)
)

tp53 &lt;- purrr::reduce(list(expr, cnv, mut), dplyr::full_join, by = &quot;sample&quot;)</code></pre>
<p>最后简单画个图吧：</p>
<pre class="r"><code>GGally::ggpairs(tp53[, -1])
#&gt; Registered S3 method overwritten by &#39;GGally&#39;:
#&gt;   method from   
#&gt;   +.gg   ggplot2
#&gt; Warning: Removed 1370 rows containing non-finite values (stat_density).
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 11009 rows containing missing values
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 12038 rows containing missing values
#&gt; Warning: Removed 11009 rows containing missing values (geom_point).
#&gt; Warning: Removed 9656 rows containing non-finite values (stat_density).
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 11608 rows containing missing values
#&gt; Warning: Removed 12038 rows containing missing values (geom_point).
#&gt; Warning: Removed 11608 rows containing missing values (geom_point).
#&gt; Warning: Removed 11397 rows containing non-finite values (stat_density).</code></pre>
<p><img src="../../../cn/post/2020-09-26-access-gene-level-cancer-data-from-ucsc-xena_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>


  <footer>
  
<nav class="post-nav">
  <span class="nav-prev">&larr; <a href="../../../cn/post/2020-09-25-fit-scatter-with-lm/">使用 ggplot2 进行曲线拟合</a></span>
  <span class="nav-next"><a href="../../../cn/post/2020-09-27-share-a-post-about-correlation/">「分享」相关分析</a> &rarr;</span>
</nav>
<script type="text/javascript">
document.addEventListener('keyup', function(e) {
  if (e.target.nodeName.toUpperCase() != 'BODY') return;
  var url = false;
  if (e.which == 37) {  
    
    url = '\/cn\/post\/2020-09-25-fit-scatter-with-lm\/';
    
  } else if (e.which == 39) {  
    
    url = '\/cn\/post\/2020-09-27-share-a-post-about-correlation\/';
    
  }
  if (url) window.location = url;
});
</script>





<div id="gitalk-container"></div>
<link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
<script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>
<script src="https://cdn.bootcss.com/blueimp-md5/2.10.0/js/md5.js"></script>
<script>
  const gitalk = new Gitalk({
    clientID: 'eed477936be7b88a8455',
    clientSecret: '32b5dde22f3c95c4fad400a31632e34221b07e83',
    repo: 'home',
    owner: 'ShixiangWang',
    admin: ['ShixiangWang'],
    id: md5(window.location.pathname), 
    distractionFreeMode: false 
  });
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('gitalk-container').innerHTML = 'Gitalk comments not available by default when the website is previewed locally.';
      return;
    }
    gitalk.render('gitalk-container');
  })();
</script>





<script async src="../../../js/fix-toc.js"></script>

<script async src="../../../js/center-img.js"></script>

<script async src="../../../js/right-quote.js"></script>

<script async src="../../../js/no-highlight.js"></script>

<script async src="../../../js/fix-footnote.js"></script>

<script async src="../../../js/math-code.js"></script>

<script async src="../../../js/external-link.js"></script>

<script async src="../../../js/alt-title.js"></script>

<script async src="../../../js/header-link.js"></script>


<script src="//yihui.org/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script async src="//yihui.org/js/center-img.js"></script>

  



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>




  
  
  

  <div class="copyright"><a href="mailto:w_shixiang@163.com"><i class='far fa-envelope fa-1x'></i></a> · <a href="https://github.com/ShixiangWang"><i class='fab fa-github fa-1x'></i></a> · <a href="https://stackoverflow.com/users/7662327/shixiang-wang"><i class='fab fa-stack-overflow fa-1x'></i></a> · <a href="https://scholar.google.com/citations?user=FvNp0NkAAAAJ&amp;hl=zh-CN"><i class='ai ai-google-scholar ai-1x'></i></a> · <a href="https://orcid.org/0000-0001-9855-7357"><i class='ai ai-orcid ai-1x'></i></a> · <a href="https://www.researchgate.net/profile/Wang_Shixiang4"><i class='ai ai-researchgate ai-1x'></i></a> <br> 本站由 <a href="https://gohugo.io">Hugo</a> 和 <a href="https://bookdown.org/yihui/blogdown/">Blogdown</a> 强力驱动 © <a href="../../../">王诗翔</a> 2017 - 2020 </div>
  
  

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/6.js?i=56h9es09xn7&amp;m=6&amp;c=ff0000&amp;cr1=ffffff&amp;f=arial&amp;l=1" async="async"></script>
  </footer>
  </article>
  
  </body>
</html>

