---
title: "一个函数获取 TCGA/CCLE 单基因分子数据"
author: "王诗翔"
date: "2020-09-26"
lastmod: "2020-09-26"
slug: ""
# All available categories:
# bioinformatics, config, docker, golang, life, linux, ml, r, read, shell, thinking
categories: ["r"]
tags: ["r", "单基因分析", "Xena", "cancer", "genomics"]
---



<blockquote>
<p>这篇文章很久之前就想写了，但一直因为一些事情拖延了。</p>
</blockquote>
<p>在开发 UCSCXenaShiny 的基础上，我将其中支持的 UCSCXena TCGA/CCLE 单基因数据下载函数进行了整理，构建了一个单一的入口。这样即使用户无需加载 Shiny，也能够简单自在的下载 癌症单基因数据了。</p>
<blockquote>
<p>这里单独说的 TCGA 不太全面，实际包含了 TCGA TARGET GTEx 3 个数据库，它们是个体水平的数据。而 CCLE 是细胞水平数据。</p>
</blockquote>
<div id="下载安装包" class="section level2">
<h2>下载安装包</h2>
<p>在国内我们统一推荐下载 Gitee 上的包：</p>
<pre class="r"><code>remotes::install_git(&quot;https://gitee.com/XenaShiny/UCSCXenaShiny&quot;)</code></pre>
<p>如果你已经安装 CRAN 上的 UCSCXenaShiny，也需要进行上面的操作，否则无法使用最新的函数。</p>
</div>
<div id="参数说明" class="section level2">
<h2>参数说明</h2>
<p>函数就一个 <code>query_value()</code>。</p>
<p>简单看看有哪些参数：</p>
<pre class="r"><code>library(UCSCXenaShiny)
args(query_value)
#&gt; function (identifier, data_type = c(&quot;gene&quot;, &quot;transcript&quot;, &quot;protein&quot;, 
#&gt;     &quot;mutation&quot;, &quot;cnv&quot;, &quot;methylation&quot;), database = c(&quot;toil&quot;, &quot;ccle&quot;)) 
#&gt; NULL</code></pre>
<p>非常简单哈，只有 3 个：</p>
<ul>
<li>第 1 个是基因名/蛋白名等。</li>
<li>第 2 个是数据的类型，包括基因表达（gene），转录本表达（transcript）、突变（mutation）、拷贝数变异（cnv）和甲基化（methylation），默认是基因表达。</li>
<li>第 3 个是数据库，包括 toil（包括上面提到的 TCGA 等几个个体水平数据）和 ccle。</li>
</ul>
</div>
<div id="使用" class="section level2">
<h2>使用</h2>
<p>了解函数参数后，使用就根据自己所需就行了。如果还不懂，可以不断试错。</p>
<p>我们以 TP53 基因为例下载一些数据看看。</p>
<pre class="r"><code>gene_expr &lt;- query_value(&quot;TP53&quot;)
#&gt; Running mode: client
#&gt; =========================================================================================
#&gt; UCSCXenaTools version 1.3.3
#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools
#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html
#&gt; 
#&gt; If you use it in published research, please cite:
#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data
#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.
#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627
#&gt; =========================================================================================
#&gt;                               --Enjoy it--
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. ProbeMap is found.
#&gt; Running mode: client
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;TcgaTargetGtex_rsem_gene_tpm&quot;) %&gt;% XenaBrowse()</code></pre>
<p>这个返回结果的结构：</p>
<pre class="r"><code>str(gene_expr)
#&gt; List of 2
#&gt;  $ expression: Named num [1:19131] 4.79 5.89 5.52 4.43 2.38 ...
#&gt;   ..- attr(*, &quot;names&quot;)= chr [1:19131] &quot;GTEX-S4Q7-0003-SM-3NM8M&quot; &quot;TCGA-19-1787-01&quot; &quot;TCGA-S9-A7J2-01&quot; &quot;GTEX-QV31-1626-SM-2S1QC&quot; ...
#&gt;  $ unit      : chr &quot;log2(tpm+0.001)&quot;</code></pre>
<p>可以查看部分数据：</p>
<pre class="r"><code>gene_expr$expression[1:5]
#&gt; GTEX-S4Q7-0003-SM-3NM8M         TCGA-19-1787-01         TCGA-S9-A7J2-01 
#&gt;                    4.79                    5.89                    5.52 
#&gt; GTEX-QV31-1626-SM-2S1QC         TCGA-G3-A3CH-11 
#&gt;                    4.43                    2.38</code></pre>
<p>有了这个数据就可以结合病人的各种表型去做分析啦。</p>
<p>我们再看下它的 CNV 和 突变情况。</p>
<pre class="r"><code># CNV
gene_cnv &lt;- query_value(&quot;TP53&quot;, data_type = &quot;cnv&quot;)
#&gt; Running mode: client
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. No probeMap found or error happened, use old way...
#&gt; Running mode: client
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;TCGA.PANCAN.sampleMap/Gistic2_CopyNumber_Gistic2_all_thresholded.by_genes&quot;) %&gt;% XenaBrowse()
gene_cnv$data[1:5]
#&gt; TCGA-A5-A0GI-01 TCGA-S9-A7J2-01 TCGA-06-0150-01 TCGA-AR-A1AH-01 TCGA-EK-A2RE-01 
#&gt;               0               0               0              -1               0

# 突变
gene_mut &lt;- query_value(&quot;TP53&quot;, data_type = &quot;mutation&quot;)
#&gt; More info about dataset please run following commands:
#&gt;   library(UCSCXenaTools)
#&gt;   XenaGenerate(subset = XenaDatasets == &quot;mc3.v0.2.8.PUBLIC.nonsilentGene.xena&quot;) %&gt;% XenaBrowse()
#&gt; Running mode: client
#&gt; Try querying data #1
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. ProbeMap is found.
#&gt; Running mode: client
gene_mut[1:5]
#&gt; TCGA-02-0003-01 TCGA-02-0033-01 TCGA-02-0047-01 TCGA-02-0055-01 TCGA-02-2470-01 
#&gt;               1               1               0               1               0</code></pre>
<p>简单利用 tidyverse 包，我们可以将几种分子数据整合到一起：</p>
<pre class="r"><code>library(tidyverse)
#&gt; ── Attaching packages ──────────────────────────────────────────────────────────── tidyverse 1.3.0 ──
#&gt; ✓ ggplot2 3.3.2     ✓ purrr   0.3.4
#&gt; ✓ tibble  3.0.3     ✓ dplyr   1.0.2
#&gt; ✓ tidyr   1.1.2     ✓ stringr 1.4.0
#&gt; ✓ readr   1.3.1     ✓ forcats 0.5.0
#&gt; ── Conflicts ─────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
#&gt; x dplyr::filter() masks stats::filter()
#&gt; x dplyr::lag()    masks stats::lag()

expr &lt;- dplyr::tibble(
  sample = names(gene_expr$expression),
  expr = as.numeric(gene_expr$expression)
)

cnv &lt;- dplyr::tibble(
  sample = names(gene_cnv$data),
  cnv = as.numeric(gene_cnv$data)
)

mut &lt;- dplyr::tibble(
  sample = names(gene_mut),
  mut = as.numeric(gene_mut)
)

tp53 &lt;- purrr::reduce(list(expr, cnv, mut), dplyr::full_join, by = &quot;sample&quot;)</code></pre>
<p>最后简单画个图吧：</p>
<pre class="r"><code>GGally::ggpairs(tp53[, -1])
#&gt; Registered S3 method overwritten by &#39;GGally&#39;:
#&gt;   method from   
#&gt;   +.gg   ggplot2
#&gt; Warning: Removed 1370 rows containing non-finite values (stat_density).
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 11009 rows containing missing values
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 12038 rows containing missing values
#&gt; Warning: Removed 11009 rows containing missing values (geom_point).
#&gt; Warning: Removed 9656 rows containing non-finite values (stat_density).
#&gt; Warning in ggally_statistic(data = data, mapping = mapping, na.rm = na.rm, :
#&gt; Removed 11608 rows containing missing values
#&gt; Warning: Removed 12038 rows containing missing values (geom_point).
#&gt; Warning: Removed 11608 rows containing missing values (geom_point).
#&gt; Warning: Removed 11397 rows containing non-finite values (stat_density).</code></pre>
<p><img src="/cn/post/2020-09-26-access-gene-level-cancer-data-from-ucsc-xena_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
