---
title: "Curriculum Vitae"
disable_comments: yes
date: '2016-01-22'
lastmod: "2020-08-02"
output:
  pdf_document:
    includes:
      in_header: cv-en-header.tex
    latex_engine: xelatex
    md_extensions: -autolink_bare_uris+hard_line_breaks
    template: cv-en-template.latex
  html_document:
    df_print: paged
geometry: margin=1in
pdf: ../Shixiang.pdf
type: cv
urlcolor: blue
fontsize: 10pt
---



<div id="shixiang-wang" class="section level1">
<h1>Shixiang Wang</h1>
<ul>
<li>Add: 233, Haike Rd., Pudong District, Shanghai 201210, China</li>
<li>Working email: <a href="mailto:wangshx@shanghaitech.edu.cn" class="email">wangshx@shanghaitech.edu.cn</a></li>
<li>Personal email: <a href="mailto:w_shixiang@163.com" class="email">w_shixiang@163.com</a> or <a href="mailto:shixiang1994wang@gmail.com" class="email">shixiang1994wang@gmail.com</a></li>
<li>Personal website: <a href="https://shixiangwang.github.io" class="uri">https://shixiangwang.github.io</a></li>
</ul>
<div id="education" class="section level2">
<h2>Education</h2>
<ul>
<li><p>2016.09 ~ Present, Ph.D Student, Cancer Biology (focusing on cancer informatics), <br>
ShanghaiTech. University, Shanghai, China</p></li>
<li><p>2012.09 ~ 2016.07, B.E. Biomedical Engineering, <br>
University of Electronic Science and Technology of China, Chengdu, China</p></li>
</ul>
</div>
<div id="professional-skills" class="section level2">
<h2>Professional skills</h2>
<ul>
<li>Programming levels:
<ul>
<li>R <span class="math inline">\(\star\star\star\star\star\)</span></li>
<li>Shell <span class="math inline">\(\star\star\star\)</span></li>
<li>Python <span class="math inline">\(\star\star\star\)</span></li>
<li>Golang <span class="math inline">\(\star\star\)</span></li>
</ul></li>
<li>Data analysis. I have advanced experience in using R and Shell for data preprocessing, data cleaning and data interpretation.</li>
<li>Statistics. I have moderate experience in using R for statistical modeling and data visualization.</li>
<li>Package/pipeline development. I master developing pure R packages and have a little experience in Python package and R Shiny development. I can combine multiple languages to create analysis pipeline.</li>
<li>Genomic analysis. I can process raw genomic data and analyze them. I have moderate experience in somatic variant calling (including SNV, INDEL and CNV), differential expression analysis and enrichment analysis.</li>
<li>Clinical analysis. I know how to construct survival models and interpret results.</li>
<li>Machine learning. I know how to do machine learning (including deep learning) and have applied some technologies to my projects.</li>
<li>Writing. I like to write with R Markdown (including Markdown) and share my knowledge to others in many ways (e.g. <a href="https://github.com/ShixiangWang/MessageBoard/issues?q=is%3Aissue+is%3Aclosed">GitHub Issue</a>, <a href="https://www.jianshu.com/u/b6608e27dc74">Jianshu</a>, <a href="https://weixin.sogou.com/weixin?type=2&amp;query=%E4%BC%98%E9%9B%85R&amp;ie=utf8&amp;s_from=input&amp;_sug_=n&amp;_sug_type_=&amp;w=01019900&amp;sut=2588&amp;sst0=1590423588976&amp;lkt=2%2C1590423588587%2C1590423588875">Wechat</a>, and etc.).</li>
</ul>
</div>
<div id="developments" class="section level2">
<h2>Developments</h2>
<ul>
<li><strong>sigminer</strong> (<a href="https://cran.r-project.org/package=sigminer" class="uri">https://cran.r-project.org/package=sigminer</a>): mutational signature analysis and visualization in R.</li>
<li><strong>ezcox</strong> (<a href="https://cran.r-project.org/package=ezcox" class="uri">https://cran.r-project.org/package=ezcox</a>): operate a batch of univariate or multivariate Cox models and return tidy result.</li>
<li><strong>DoAbsolute</strong> (<a href="https://github.com/ShixiangWang/DoAbsolute" class="uri">https://github.com/ShixiangWang/DoAbsolute</a>): automate ABSOLUTE calling for multiple samples in parallel way.</li>
<li><strong>metawho</strong> (<a href="https://cran.r-project.org/package=metawho" class="uri">https://cran.r-project.org/package=metawho</a>): simple R implementation of “<a href="https://www.bmj.com/content/356/bmj.j573">Meta-analytical method to Identify Who Benefits Most from Treatments</a>”.</li>
<li><strong>UCSCXenaTools</strong> (<a href="https://cran.r-project.org/package=UCSCXenaTools" class="uri">https://cran.r-project.org/package=UCSCXenaTools</a>): an R package for downloading and exploring data from <a href="https://xenabrowser.net/datapages/">UCSC Xena data hubs</a>.</li>
<li><strong>UCSCXenaShiny</strong> (<a href="https://cran.r-project.org/package=UCSCXenaShiny" class="uri">https://cran.r-project.org/package=UCSCXenaShiny</a>): a Shiny based on UCSCXenaTools.</li>
<li><strong>contribution</strong> (<a href="https://cran.r-project.org/package=contribution" class="uri">https://cran.r-project.org/package=contribution</a>): generate contribution table for credit assignment in a project.</li>
<li><strong>loon</strong> (<a href="https://pypi.org/project/loon/" class="uri">https://pypi.org/project/loon/</a>): a Python toolkit for operating remote host based on SSH.</li>
<li><strong>sync-deploy</strong> (<a href="https://github.com/ShixiangWang/sync-deploy" class="uri">https://github.com/ShixiangWang/sync-deploy</a>): a Shell toolkit for deploying script/command task on a remote host, including up/download files, run script and more.</li>
</ul>
<p>More activities about my development and contribution can be viewed at <a href="https://github.com/ShixiangWang">github.com/ShixiangWang</a>.</p>
</div>
<div id="publications" class="section level2">
<h2>Publications</h2>
<p>Total citations: 106. (Data source: <a href="https://scholar.google.com/citations?user=FvNp0NkAAAAJ">Google Scholar</a>. Update time: 2020-08-02)</p>
<ul>
<li><strong>Wang, S.</strong>, He, Z., Wang, X., Li, H., &amp; Liu, X. S. (2019). Antigen presentation and tumor immunogenicity in cancer immunotherapy response prediction. <em>eLife</em>, 8, e49020. <a href="https://doi.org/10.7554/eLife.49020" class="uri">https://doi.org/10.7554/eLife.49020</a> <i class="fa fa-file-pdf-o"> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6879305/pdf/elife-49020.pdf">(PDF)</a></i></li>
<li><strong>Wang, S.</strong>, He, Z., Wang, X., Li, H., Wu, T., Sun, X., … &amp; Liu, X. S. (2019). Can tumor mutational burden determine the most effective treatment for lung cancer patients?. <em>Lung Cancer Management</em>. <a href="https://doi.org/10.2217/lmt-2019-0013" class="uri">https://doi.org/10.2217/lmt-2019-0013</a> <i class="fa fa-file-pdf-o"> <a href="https://www.futuremedicine.com/doi/pdf/10.2217/lmt-2019-0013">(PDF)</a></i></li>
<li><strong>Wang, S.</strong>, Cowley, L. A., &amp; Liu, X. S. (2019). Sex differences in Cancer immunotherapy efficacy, biomarkers, and therapeutic strategy. <em>Molecules</em>, 24(18), 3214. <i class="fa fa-file-pdf-o"> <a href="https://www.mdpi.com/1420-3049/24/18/3214/pdf">(PDF)</a></i></li>
<li><strong>Wang, S.</strong> &amp; Liu, X. S. (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. <em>Journal of Open Source Software</em>, 4(40), 1627, <a href="https://doi.org/10.21105/joss.01627" class="uri">https://doi.org/10.21105/joss.01627</a> <i class="fa fa-file-pdf-o"> <a href="https://www.theoj.org/joss-papers/joss.01627/10.21105.joss.01627.pdf">(PDF)</a></i></li>
<li>He, Z., <strong>Wang, S.</strong>, Shao, Y., Zhang, J., Wu, X., Chen, Y., … &amp; Liu, X. S. (2019). Ras downstream effector GGCT alleviates oncogenic stress. <em>iScience</em>. <i class="fa fa-file-pdf-o"> <a href="https://www.researchgate.net/publication/334723955_Ras_downstream_effector_GGCT_alleviates_oncogenic_stress">(PDF)</a></i></li>
<li><strong>Wang, S.</strong>, Zhang, J., He, Z., Wu, K., &amp; Liu, X. S. (2019). The predictive power of tumor mutational burden in lung cancer immunotherapy response is influenced by patients’ sex. <em>International journal of cancer</em>, 145(10), 2840-2849. <i class="fa fa-file-pdf-o"> <a href="https://www.researchgate.net/publication/332372671_The_predictive_power_of_tumor_mutational_burden_in_lung_cancer_immunotherapy_response_is_influenced_by_patients&#39;_sex">(PDF)</a></i></li>
<li><strong>Wang, S.</strong>, Jia, M., He, Z., &amp; Liu, X. S. (2018). APOBEC3B and APOBEC mutational signature as potential predictive markers for immunotherapy response in non-small cell lung cancer. <em>Oncogene</em>, 37(29), 3924-3936. <i class="fa fa-file-pdf-o"> <a href="https://www.researchgate.net/publication/324781475_APOBEC3B_and_APOBEC_mutational_signature_as_potential_predictive_markers_for_immunotherapy_response_in_non-small_cell_lung_cancer">(PDF)</a></i></li>
</ul>
</div>
</div>
