---
title: "Curriculum Vitae"
disable_comments: yes
date: '2016-01-22'
lastmod: "`r Sys.Date()`"
output:
  pdf_document:
    includes:
      in_header: cv-en-header.tex
    latex_engine: xelatex
    md_extensions: -autolink_bare_uris+hard_line_breaks
    template: cv-en-template.latex
  html_document:
    df_print: paged
geometry: margin=1in
pdf: ../Shixiang.pdf
type: cv
urlcolor: blue
fontsize: 10pt
---

# Shixiang Wang

- Add: 233, Haike Rd., Pudong District, Shanghai 201210, China
- Working email: <wangshx@shanghaitech.edu.cn>
- Personal email: <w_shixiang@163.com> or <shixiang1994wang@gmail.com>
- Personal website: <https://shixiangwang.github.io>

## Education

- 2016.09 ~ Present, Ph.D Student, Cancer Biology (focusing on cancer informatics), <br>
  ShanghaiTech. University, Shanghai, China

- 2012.09 ~ 2016.07, B.E. Biomedical Engineering, <br>
  University of Electronic Science and Technology of China, Chengdu, China


## Professional skills

- Programming levels:
  - R $\star\star\star\star\star$
  - Shell $\star\star\star$
  - Python $\star\star\star$
  - Golang $\star\star$
- Data analysis. I have advanced experience in using R and Shell for data preprocessing, data cleaning and data interpretation.
- Statistics. I have moderate experience in using R for statistical modeling and data visualization. 
- Package/pipeline development. I master developing pure R packages and have a little experience in Python package and R Shiny development. I can combine multiple languages to create analysis pipeline.
- Genomic analysis. I can process raw genomic data and analyze them. I have moderate experience in somatic variant calling (including SNV, INDEL and CNV), differential expression analysis and enrichment analysis.
- Clinical analysis. I know how to construct survival models and interpret results.
- Machine learning. I know how to do machine learning (including deep learning) and have applied some technologies to my projects.
- Writing. I like to write with R Markdown (including Markdown) and share my knowledge to others in many ways (e.g. [GitHub Issue](https://github.com/ShixiangWang/MessageBoard/issues?q=is%3Aissue+is%3Aclosed), [Jianshu](https://www.jianshu.com/u/b6608e27dc74), [Wechat](https://weixin.sogou.com/weixin?type=2&query=%E4%BC%98%E9%9B%85R&ie=utf8&s_from=input&_sug_=n&_sug_type_=&w=01019900&sut=2588&sst0=1590423588976&lkt=2%2C1590423588587%2C1590423588875), and etc.). 

## Developments

- **sigminer** (<https://cran.r-project.org/package=sigminer>): mutational signature analysis and visualization in R.
- **ezcox** (<https://cran.r-project.org/package=ezcox>): operate a batch of univariate or multivariate Cox models and return tidy result.
- **DoAbsolute** (<https://github.com/ShixiangWang/DoAbsolute>): automate ABSOLUTE calling for multiple samples in parallel way.
- **metawho** (<https://cran.r-project.org/package=metawho>): simple R implementation of “[Meta-analytical method to Identify Who Benefits Most from Treatments](https://www.bmj.com/content/356/bmj.j573)”.
- **UCSCXenaTools** (<https://cran.r-project.org/package=UCSCXenaTools>): an R package for downloading and exploring data from [UCSC Xena data hubs](https://xenabrowser.net/datapages/).
- **UCSCXenaShiny** (<https://cran.r-project.org/package=UCSCXenaShiny>): a Shiny based on UCSCXenaTools.
- **contribution** (<https://cran.r-project.org/package=contribution>): generate contribution table for credit assignment in a project.
- **loon** (<https://pypi.org/project/loon/>): a Python toolkit for operating remote host based on SSH.
- **sync-deploy** (<https://github.com/ShixiangWang/sync-deploy>): a Shell toolkit for deploying script/command task on a remote host, including up/download files, run script and more.

More activities about my development and contribution can be viewed at [github.com/ShixiangWang](https://github.com/ShixiangWang).

## Publications

```{r, include=FALSE, eval=FALSE}
# cannot use in China
library(scholar)
library(ggplot2)
id <- "FvNp0NkAAAAJ"
## Get his citation history, i.e. citations to his work in a given year
ct <- get_citation_history(id)
ggplot(ct, aes(year, cites)) + geom_line() + geom_point()
```

Total citations: 73. (Data source: [Google Scholar](https://scholar.google.com/citations?user=FvNp0NkAAAAJ). Update time: `r Sys.Date()`)

- **Wang, S.**, He, Z., Wang, X., Li, H., & Liu, X. S. (2019). Antigen presentation and tumor immunogenicity in cancer immunotherapy response prediction. *eLife*, 8, e49020. <https://doi.org/10.7554/eLife.49020> <i class="fa fa-file-pdf-o"> [(PDF)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6879305/pdf/elife-49020.pdf)</i> 
- **Wang, S.**, He, Z., Wang, X., Li, H., Wu, T., Sun, X., ... & Liu, X. S. (2019). Can tumor mutational burden determine the most effective treatment for lung cancer patients?. *Lung Cancer Management*. <https://doi.org/10.2217/lmt-2019-0013> <i class="fa fa-file-pdf-o"> [(PDF)](https://www.futuremedicine.com/doi/pdf/10.2217/lmt-2019-0013)</i> 
- **Wang, S.**, Cowley, L. A., & Liu, X. S. (2019). Sex differences in Cancer immunotherapy efficacy, biomarkers, and therapeutic strategy. *Molecules*, 24(18), 3214. <i class="fa fa-file-pdf-o"> [(PDF)](https://www.mdpi.com/1420-3049/24/18/3214/pdf)</i> 
- **Wang, S.** & Liu, X. S. (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. *Journal of Open Source Software*, 4(40), 1627, https://doi.org/10.21105/joss.01627 <i class="fa fa-file-pdf-o"> [(PDF)](https://www.theoj.org/joss-papers/joss.01627/10.21105.joss.01627.pdf)</i>
- He, Z., **Wang, S.**, Shao, Y., Zhang, J., Wu, X., Chen, Y., ... & Liu, X. S. (2019). Ras downstream effector GGCT alleviates oncogenic stress. *iScience*. <i class="fa fa-file-pdf-o"> [(PDF)](https://www.researchgate.net/publication/334723955_Ras_downstream_effector_GGCT_alleviates_oncogenic_stress)</i>
- **Wang, S.**, Zhang, J., He, Z., Wu, K., & Liu, X. S. (2019). The predictive power of tumor mutational burden in lung cancer immunotherapy response is influenced by patients’ sex. *International journal of cancer*, 145(10), 2840-2849. <i class="fa fa-file-pdf-o"> [(PDF)](https://www.researchgate.net/publication/332372671_The_predictive_power_of_tumor_mutational_burden_in_lung_cancer_immunotherapy_response_is_influenced_by_patients'_sex)</i>
- **Wang, S.**, Jia, M., He, Z., & Liu, X. S. (2018). APOBEC3B and APOBEC mutational signature as potential predictive markers for immunotherapy response in non-small cell lung cancer. *Oncogene*, 37(29), 3924-3936. <i class="fa fa-file-pdf-o"> [(PDF)](https://www.researchgate.net/publication/324781475_APOBEC3B_and_APOBEC_mutational_signature_as_potential_predictive_markers_for_immunotherapy_response_in_non-small_cell_lung_cancer)</i>



