---
slug: ucscxenatools-201908
title: 'UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis'
package_version: 1.2.10
authors:
  - Shixiang Wang
date: '2019-08-25'
categories: [bioinformatics]
tags: ["r", "software", "package", "bioinformatics", "data-access", "survival-analysis", "UCSCXenaTools"]
lastmod: "2020-08-09"
header-includes:
- \usepackage{caption}
- \captionsetup[figure]{labelformat=empty}
---





<blockquote>
<p>I thank the edition made by <a href="https://github.com/stefaniebutland">Stefanie Butland</a>.</p>
</blockquote>
<p>The <a href="https://xenabrowser.net/">UCSC Xena platform</a> provides an unprecedented resource for public omics data from big projects like <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">The Cancer Genome Atlas (TCGA)</a>, however, it is hard
for users to incorporate multiple datasets or data types, integrate the selected data with
popular analysis tools or homebrewed code, and reproduce analysis procedures. To address this issue, we developed an R package UCSCXenaTools for enabling data retrieval, analysis integration and reproducible research for omics data from the UCSC Xena platform<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<p>In this technote we will outline how to use the UCSCXenaTools package to pull gene expression and clinical data from <a href="https://xena.ucsc.edu/">UCSC Xena</a> for survival analysis.</p>
<p>For general usage of UCSCXenaTools, please refer to the <a href="https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html">package vignette</a>. Any bug or feature request can be reported in <a href="https://github.com/ropensci/UCSCXenaTools/issues">GitHub issues</a>.</p>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p><strong>UCSCXenaTools</strong> is available from CRAN:</p>
<pre class="r"><code>install.packages(&quot;UCSCXenaTools&quot;)</code></pre>
<p>Alternatively, the latest development version can be downloaded from GitHub:</p>
<pre class="r"><code>remotes::install_github(&quot;ropensci/UCSCXenaTools&quot;, build_vignettes = TRUE, dependencies = TRUE)</code></pre>
</div>
<div id="how-it-works" class="section level2">
<h2>How it works</h2>
<p>Before actually pulling data, understanding how UCSCXenaTools works (see Figure 1) will help users locate the most important function to use.</p>
<p>Generally,</p>
<ul>
<li>for operating datasets, we use functions whose names start with <code>Xena</code></li>
<li>for operating subset of a dataset, we use functions whose names start with <code>fetch_</code></li>
</ul>
<p><img src="https://raw.githubusercontent.com/ropensci/UCSCXenaTools/master/paper/overview.png" />
<em>Figure 1. The UCSCXenaTools pipeline</em></p>
<p>We will provide an example illustrating how to use UCSCXenaTools to study the effect of expression of the <a href="https://ghr.nlm.nih.gov/gene/KRAS"><em>KRAS</em></a> gene on prognosis of Lung Adenocarcinoma (LUAD) patients. <em>KRAS</em> is a known driver gene in LUAD. We retrieve expression data for the <em>KRAS</em> gene and survival status data for <a href="https://xenabrowser.net/datapages/?cohort=TCGA%20Lung%20Adenocarcinoma%20(LUAD)&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">LUAD patients from the TCGA</a> and use these as input to a survival analysis, frequently used in cancer research.</p>
</div>
<div id="download-data" class="section level2">
<h2>Download data</h2>
<p>First we get information on all datasets in the TCGA LUAD cohort and store as <code>luad_cohort</code> object.</p>
<pre class="r"><code>suppressMessages(library(UCSCXenaTools))
suppressMessages(library(dplyr))</code></pre>
<pre class="r"><code>luad_cohort = XenaData %&gt;% 
  filter(XenaHostNames == &quot;tcgaHub&quot;) %&gt;% # select TCGA Hub
  XenaScan(&quot;TCGA Lung Adenocarcinoma&quot;)   # select LUAD cohort

luad_cohort
#&gt; # A tibble: 23 x 17
#&gt;    XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype
#&gt;    &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;      
#&gt;  1 https://… tcgaHub       TCGA Lung … RABIT/separ…         467 pathway ac…
#&gt;  2 https://… tcgaHub       TCGA Lung … RABIT/separ…         120 pathway ac…
#&gt;  3 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         151 DNA methyl…
#&gt;  4 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         492 DNA methyl…
#&gt;  5 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         516 copy numbe…
#&gt;  6 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         237 protein ex…
#&gt;  7 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         576 gene expre…
#&gt;  8 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…          60 miRNA matu…
#&gt;  9 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         576 gene expre…
#&gt; 10 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         576 gene expre…
#&gt; # … with 13 more rows, and 11 more variables: Label &lt;chr&gt;, Type &lt;chr&gt;,
#&gt; #   AnatomicalOrigin &lt;chr&gt;, SampleType &lt;chr&gt;, Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;,
#&gt; #   LongTitle &lt;chr&gt;, Citation &lt;chr&gt;, Version &lt;chr&gt;, Unit &lt;chr&gt;, Platform &lt;chr&gt;</code></pre>
<div id="download-clinical-dataset" class="section level3">
<h3>Download clinical dataset</h3>
<p>Now we download clinical datasets of TCGA LUAD cohort and load them into R.</p>
<pre class="r"><code>cli_query = luad_cohort %&gt;% 
  filter(DataSubtype == &quot;phenotype&quot;) %&gt;%  # select clinical dataset
  XenaGenerate() %&gt;%  # generate a XenaHub object
  XenaQuery() %&gt;% 
  XenaDownload()
#&gt; This will check url status, please be patient.
#&gt; All downloaded files will under directory /var/folders/bj/nw1w4g1j37ddpgb6zmh3sfh80000gn/T//RtmpLpcBQu.
#&gt; The &#39;trans_slash&#39; option is FALSE, keep same directory structure as Xena.
#&gt; Creating directories for datasets...
#&gt; Downloading TCGA.LUAD.sampleMap/LUAD_clinicalMatrix
#&gt; Downloading survival/LUAD_survival.txt.gz

cli = XenaPrepare(cli_query)</code></pre>
<blockquote>
<p>NOTE: from 2019-12-06, UCSC Xena stores phenotype data in two
datasets, one is survival data, the other is other phenotype data.
So <code>cli</code> is a list here.</p>
</blockquote>
<p>Keep survival data here, we only need it.</p>
<pre class="r"><code>cli = cli$LUAD_survival.txt.gz

# See a few rows
head(cli)
#&gt; # A tibble: 6 x 11
#&gt;   sample `_PATIENT`    OS OS.time   DSS DSS.time   DFI DFI.time   PFI PFI.time
#&gt;   &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;
#&gt; 1 TCGA-… TCGA-05-4…     0       0     0        0    NA       NA     0        0
#&gt; 2 TCGA-… TCGA-05-4…     0    1523     0     1523    NA       NA     0     1523
#&gt; 3 TCGA-… TCGA-05-4…     1     121    NA      121    NA       NA     0      121
#&gt; 4 TCGA-… TCGA-05-4…     0     607     0      607     1      334     1      334
#&gt; 5 TCGA-… TCGA-05-4…     0     426     0      426    NA       NA     1      183
#&gt; 6 TCGA-… TCGA-05-4…     0    1369     0     1369    NA       NA     0     1369
#&gt; # … with 1 more variable: Redaction &lt;lgl&gt;</code></pre>
</div>
<div id="download-kras-gene-expression" class="section level3">
<h3>Download <em>KRAS</em> gene expression</h3>
<p>To download gene expression data, first we need to select the right dataset.</p>
<pre class="r"><code>ge = luad_cohort %&gt;% 
  filter(DataSubtype == &quot;gene expression RNAseq&quot;, Label == &quot;IlluminaHiSeq&quot;)
ge
#&gt; # A tibble: 1 x 17
#&gt;   XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype Label
#&gt;   &lt;chr&gt;     &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;              &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;
#&gt; 1 https://… tcgaHub       TCGA Lung … TCGA.LUAD.s…         576 gene expre… Illu…
#&gt; # … with 10 more variables: Type &lt;chr&gt;, AnatomicalOrigin &lt;chr&gt;,
#&gt; #   SampleType &lt;chr&gt;, Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;, LongTitle &lt;chr&gt;,
#&gt; #   Citation &lt;chr&gt;, Version &lt;chr&gt;, Unit &lt;chr&gt;, Platform &lt;chr&gt;</code></pre>
<p>Now we fetch <em>KRAS</em> gene expression values.</p>
<pre class="r"><code># You can pass gene symbols to &#39;identifiers&#39; option
# to obtain their values in a dataset.
# A matrix will be returned by &#39;fetch_dense_values&#39; function
# with rows corresponding to genes,
# so here we extract the first row.
KRAS = fetch_dense_values(host = ge$XenaHosts, 
                          dataset = ge$XenaDatasets, 
                          identifiers = &quot;KRAS&quot;,
                          use_probeMap = TRUE) %&gt;% 
  .[1, ]
#&gt; -&gt; Checking identifiers...
#&gt; -&gt; use_probeMap is TRUE, skipping checking identifiers...
#&gt; -&gt; Done.
#&gt; -&gt; Checking samples...
#&gt; -&gt; Done.
#&gt; -&gt; Checking if the dataset has probeMap...
#&gt; -&gt; Done. ProbeMap is found.

head(KRAS)
#&gt; TCGA-69-7978-01 TCGA-62-8399-01 TCGA-78-7539-01 TCGA-50-5931-11 TCGA-73-4658-01 
#&gt;           10.25           10.29           10.82           10.29           10.36 
#&gt; TCGA-44-6775-01 
#&gt;           10.03</code></pre>
</div>
<div id="merge-expression-data-and-survival-status" class="section level3">
<h3>Merge expression data and survival status</h3>
<p>Next, we join the two <code>data.frame</code> by <code>sample</code> and keep necessary columns.
Here we focus on ‘Primary Tumor’ for simplicity.</p>
<pre class="r"><code>merged_data = tibble(sample = names(KRAS),
                     KRAS_expression = as.numeric(KRAS)) %&gt;% 
  left_join(cli, by = &quot;sample&quot;) %&gt;% 
  filter(substr(sample, 14, 15) == &quot;01&quot;) %&gt;%  # Keep only &#39;Primary Tumor&#39;
  select(sample, KRAS_expression, OS.time, OS) %&gt;% 
  rename(time = OS.time, 
         status = OS)
  

head(merged_data)
#&gt; # A tibble: 6 x 4
#&gt;   sample          KRAS_expression  time status
#&gt;   &lt;chr&gt;                     &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;
#&gt; 1 TCGA-69-7978-01           10.2    134      0
#&gt; 2 TCGA-62-8399-01           10.3   2696      0
#&gt; 3 TCGA-78-7539-01           10.8    791      0
#&gt; 4 TCGA-73-4658-01           10.4   1600      1
#&gt; 5 TCGA-44-6775-01           10.0    705      0
#&gt; 6 TCGA-44-2655-01            9.75  1324      0</code></pre>
</div>
</div>
<div id="survival-analysis" class="section level2">
<h2>Survival analysis</h2>
<p>To study the effect of <em>KRAS</em> gene expression on prognosis of LUAD patients, we show two approaches:</p>
<ol style="list-style-type: decimal">
<li>use Cox model to determine the effect when <em>KRAS</em> gene expression increases</li>
<li>use Kaplan-Meier curve and log-rank test to observe the difference in different of<em>KRAS</em> gene expression status, i.e. high or low</li>
</ol>
<p>We will use package <a href="https://cran.r-project.org/web/packages/survival/index.html"><strong>survival</strong></a> and <a href="https://cran.r-project.org/web/packages/survminer/index.html"><strong>survminer</strong></a> to create models and plot survival curves, respectively.</p>
<pre class="r"><code>library(survival)
library(survminer)
#&gt; Loading required package: ggplot2
#&gt; Loading required package: ggpubr</code></pre>
<div id="cox-model" class="section level3">
<h3>Cox model</h3>
<pre class="r"><code>fit = coxph(Surv(time, status) ~ KRAS_expression, data = merged_data)
fit
#&gt; Call:
#&gt; coxph(formula = Surv(time, status) ~ KRAS_expression, data = merged_data)
#&gt; 
#&gt;                   coef exp(coef) se(coef)     z       p
#&gt; KRAS_expression 0.2922    1.3394   0.1020 2.864 0.00418
#&gt; 
#&gt; Likelihood ratio test=7.64  on 1 df, p=0.0057
#&gt; n= 506, number of events= 183 
#&gt;    (9 observations deleted due to missingness)</code></pre>
<p>We can find that patients with higher <em>KRAS</em> gene expression have higher risk (34% increase per <em>KRAS</em> gene expression unit increase), and the effect of <em>KRAS</em> gene expression is statistically significant (<em>p</em>&lt;0.05).</p>
<blockquote>
<p>If you know little about survival analysis, two blogs are recommended to read:</p>
<ol style="list-style-type: decimal">
<li><a href="http://www.sthda.com/english/wiki/survival-analysis-basics">Survival Analysis Basics</a></li>
<li><a href="http://www.sthda.com/english/wiki/cox-proportional-hazards-model">Cox Proportional-Hazards Model</a></li>
</ol>
</blockquote>
</div>
<div id="risk-between-expression-groups" class="section level3">
<h3>Risk between expression groups</h3>
<p>We can also divide patients into two groups using KRAS median as a cutoff.</p>
<pre class="r"><code>merged_data = merged_data %&gt;% 
  mutate(group = case_when(
    KRAS_expression &gt; quantile(KRAS_expression, 0.5) ~ &#39;KRAS_High&#39;,
    KRAS_expression &lt; quantile(KRAS_expression, 0.5) ~ &#39;KRAS_Low&#39;,
    TRUE ~ NA_character_
  ))

fit = survfit(Surv(time, status) ~ group, data = merged_data)</code></pre>
<p>Then we can plot the survival curves for each group.</p>
<pre class="r"><code>ggsurvplot(fit, pval = TRUE)</code></pre>
<div class="figure"><span id="fig:unnamed-chunk-14"></span>
<img src="/en/post/2019-08-20-UCSCXenaTools-notes-to-rOpenSci_files/figure-html/unnamed-chunk-14-1.png" alt="Figure 2. Kaplan-Meier curve. Survival probability vs Time (days)" width="672" />
<p class="caption">
Figure 1: Figure 2. Kaplan-Meier curve. Survival probability vs Time (days)
</p>
</div>
<p>The Kaplan-Meier plot shows what percent of patients are alive at a time point. We can clearly see that patients in ‘KRAS_Low’ group have better survival than patients in ‘KRAS_High’ group because the survival probability of ‘KRAS_High’ group is always lower than ‘KRAS_Low’ group over time (the unit is ‘day’ here). The difference between the two groups is statistically significant (<em>p</em>&lt;0.05 by log-rank test).</p>
</div>
<div id="related-project" class="section level3">
<h3>Related project</h3>
<p><a href="https://github.com/openbiox/XenaShiny"><strong>XenaShiny</strong></a>, a Shiny project based on UCSCXenaTools, is under development by my friends and me.</p>
</div>
<div id="acknowledgements" class="section level3">
<h3>Acknowledgements</h3>
<p>We thank <a href="https://github.com/cstawitz">Christine Stawitz</a> and <a href="https://github.com/carlganz">Carl Ganz</a> for their constructive comments. This package is reviewed by rOpenSci at <a href="https://github.com/ropensci/software-review/issues/315" class="uri">https://github.com/ropensci/software-review/issues/315</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software, 4(40), 1627, <a href="https://doi.org/10.21105/joss.01627" class="uri">https://doi.org/10.21105/joss.01627</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
