---
slug: ucscxenatools-201908
title: 'UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis'
package_version: 1.2.10
authors:
  - Shixiang Wang
date: '2019-08-25'
categories: [bioinformatics]
tags: ["r", "software", "package", "bioinformatics", "data-access", "survival-analysis", "UCSCXenaTools"]
lastmod: "`r Sys.Date()`"
header-includes:
- \usepackage{caption}
- \captionsetup[figure]{labelformat=empty}
---

```{r, include=FALSE}
knitr::opts_chunk$set(comment = "#>", collapse = TRUE)
Sys.setenv("LANGUAGE"="EN")
```

> I thank the edition made by [Stefanie Butland](https://github.com/stefaniebutland).

The [UCSC Xena platform](https://xenabrowser.net/) provides an unprecedented resource for public omics data from big projects like [The Cancer Genome Atlas (TCGA)](https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga), however, it is hard
for users to incorporate multiple datasets or data types, integrate the selected data with
popular analysis tools or homebrewed code, and reproduce analysis procedures. To address this issue, we developed an R package UCSCXenaTools for enabling data retrieval, analysis integration and reproducible research for omics data from the UCSC Xena platform[^1].

In this technote we will outline how to use the UCSCXenaTools package to pull gene expression and clinical data from [UCSC Xena](https://xena.ucsc.edu/) for survival analysis.

For general usage of UCSCXenaTools, please refer to the [package vignette](https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html). Any bug or feature request can be reported in [GitHub issues](https://github.com/ropensci/UCSCXenaTools/issues).

## Installation

**UCSCXenaTools** is available from CRAN:

```{r, eval=FALSE}
install.packages("UCSCXenaTools")
```

Alternatively, the latest development version can be downloaded from GitHub:

```{r, eval=FALSE}
remotes::install_github("ropensci/UCSCXenaTools", build_vignettes = TRUE, dependencies = TRUE)
```

## How it works

Before actually pulling data, understanding how UCSCXenaTools works (see Figure 1) will help users locate the most important function to use.

Generally,

- for operating datasets, we use functions whose names start with `Xena`
- for operating subset of a dataset, we use functions whose names start with `fetch_`

![](https://raw.githubusercontent.com/ropensci/UCSCXenaTools/master/paper/overview.png)
*Figure 1. The UCSCXenaTools pipeline*

We will provide an example illustrating how to use UCSCXenaTools to study the effect of expression of the [*KRAS*](https://ghr.nlm.nih.gov/gene/KRAS) gene on prognosis of Lung Adenocarcinoma (LUAD) patients. *KRAS* is a known driver gene in LUAD. We retrieve expression data for the *KRAS* gene and survival status data for [LUAD patients from the TCGA](https://xenabrowser.net/datapages/?cohort=TCGA%20Lung%20Adenocarcinoma%20(LUAD)&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443) and use these as input to a survival analysis, frequently used in cancer research.

## Download data

First we get information on all datasets in the TCGA LUAD cohort and store as `luad_cohort` object.

```{r}
suppressMessages(library(UCSCXenaTools))
suppressMessages(library(dplyr))
```

```{r}
luad_cohort = XenaData %>% 
  filter(XenaHostNames == "tcgaHub") %>% # select TCGA Hub
  XenaScan("TCGA Lung Adenocarcinoma")   # select LUAD cohort

luad_cohort
```


### Download clinical dataset

Now we download clinical datasets of TCGA LUAD cohort and load them into R.

```{r}
cli_query = luad_cohort %>% 
  filter(DataSubtype == "phenotype") %>%  # select clinical dataset
  XenaGenerate() %>%  # generate a XenaHub object
  XenaQuery() %>% 
  XenaDownload()

cli = XenaPrepare(cli_query)
```

> NOTE: from 2019-12-06, UCSC Xena stores phenotype data in two
> datasets, one is survival data, the other is other phenotype data.
> So `cli` is a list here.

Keep survival data here, we only need it.

```{r}
cli = cli$LUAD_survival.txt.gz

# See a few rows
head(cli)
```



### Download *KRAS* gene expression

To download gene expression data, first we need to select the right dataset.

```{r}
ge = luad_cohort %>% 
  filter(DataSubtype == "gene expression RNAseq", Label == "IlluminaHiSeq")
ge
```

Now we fetch *KRAS* gene expression values.

```{r}
# You can pass gene symbols to 'identifiers' option
# to obtain their values in a dataset.
# A matrix will be returned by 'fetch_dense_values' function
# with rows corresponding to genes,
# so here we extract the first row.
KRAS = fetch_dense_values(host = ge$XenaHosts, 
                          dataset = ge$XenaDatasets, 
                          identifiers = "KRAS",
                          use_probeMap = TRUE) %>% 
  .[1, ]

head(KRAS)
```

### Merge expression data and survival status

Next, we join the two `data.frame` by `sample` and keep necessary columns.
Here we focus on 'Primary Tumor' for simplicity.

```{r}
merged_data = tibble(sample = names(KRAS),
                     KRAS_expression = as.numeric(KRAS)) %>% 
  left_join(cli, by = "sample") %>% 
  filter(substr(sample, 14, 15) == "01") %>%  # Keep only 'Primary Tumor'
  select(sample, KRAS_expression, OS.time, OS) %>% 
  rename(time = OS.time, 
         status = OS)
  

head(merged_data)
```


## Survival analysis

To study the effect of *KRAS* gene expression on prognosis of LUAD patients, we show two approaches:

1. use Cox model to determine the effect when *KRAS* gene expression increases
2. use Kaplan-Meier curve and log-rank test to observe the difference in different of*KRAS* gene expression status, i.e. high or low

We will use package [**survival**](https://cran.r-project.org/web/packages/survival/index.html) and [**survminer**](https://cran.r-project.org/web/packages/survminer/index.html) to create models and plot survival curves, respectively.

```{r}
library(survival)
library(survminer)
```

### Cox model

```{r}
fit = coxph(Surv(time, status) ~ KRAS_expression, data = merged_data)
fit
```

We can find that patients with higher *KRAS* gene expression have higher risk (34% increase per *KRAS* gene expression unit increase), and the effect of *KRAS* gene expression is statistically significant (*p*<0.05).

> If you know little about survival analysis, two blogs are recommended to read:
>
> 1. [Survival Analysis Basics](http://www.sthda.com/english/wiki/survival-analysis-basics)
> 2. [Cox Proportional-Hazards Model](http://www.sthda.com/english/wiki/cox-proportional-hazards-model)

### Risk between expression groups

We can also divide patients into two groups using KRAS median as a cutoff.

```{r}
merged_data = merged_data %>% 
  mutate(group = case_when(
    KRAS_expression > quantile(KRAS_expression, 0.5) ~ 'KRAS_High',
    KRAS_expression < quantile(KRAS_expression, 0.5) ~ 'KRAS_Low',
    TRUE ~ NA_character_
  ))

fit = survfit(Surv(time, status) ~ group, data = merged_data)
```


Then we can plot the survival curves for each group.

```{r, fig.cap="Figure 2. Kaplan-Meier curve. Survival probability vs Time (days)"}
ggsurvplot(fit, pval = TRUE)
```

The Kaplan-Meier plot shows what percent of patients are alive at a time point. We can clearly see that patients in ‘KRAS_Low’ group have better survival than patients in ‘KRAS_High’ group because the survival probability of 'KRAS_High' group is always lower than 'KRAS_Low' group over time (the unit is 'day' here). The difference between the two groups is statistically significant (*p*<0.05 by log-rank test).

### Related project

[**XenaShiny**](https://github.com/openbiox/XenaShiny), a Shiny project based on UCSCXenaTools, is under development by my friends and me.

### Acknowledgements

We thank [Christine Stawitz](https://github.com/cstawitz) and [Carl Ganz](https://github.com/carlganz) for their constructive comments. This package is reviewed by rOpenSci at <https://github.com/ropensci/software-review/issues/315>.


[^1]: Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627
