---
title: "Obtain RNAseq Values for a Specific Gene in Xena Database"
author: "王诗翔"
date: "2020-07-22"
lastmod: "`r Sys.Date()`"
slug: ""
categories: [bioinformatics]
tags: [r, package, UCSCXenaTools]
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE, dev = "png", comment = "#>", tidy = "styler")
#Sys.setenv("LANGUAGE"="EN") # Embed this for outputing English message
#Sys.setlocale('LC_ALL','C') # Embed this directly in the Rmarkdown script that contains the Chinese character comment
options(digits=3)
options(max.print=200)
```


When using `UCSCXenaTools` package, you may want to focus on single gene analysis, a typical case has been shown in my previous blog [UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis](https://shixiangwang.github.io/home/en/post/ucscxenatools-201908/). Here I will describe how to get single gene values (especially RNAseq data) in details.

Let's load package.

```{r}
library(UCSCXenaTools)
```

## First, Find Your Interest Dataset

UCSC Xena provides more than 1000 datasets, when you want to get values for single gene, you must select a target dataset. You can find them in the following table or from [UCSC Xena datasets page](https://xenabrowser.net/datapages/).

```{r}
DT::datatable(UCSCXenaTools::XenaData)
```

Pick up a dataset and get its `XenaHosts` and `XenaDatasets`, i.e. get its data hub host URL and dataset ID. You can copy them or you can use your R skill to get and store them in a object. For example, I got a reader want to study RNASeq values of TCGA LUAD gene.

I can use R:

```{r}
library(dplyr)
ge = XenaData %>% 
  filter(XenaHostNames == "tcgaHub") %>% # select TCGA Hub
  XenaScan("TCGA Lung Adenocarcinoma") %>% 
  filter(DataSubtype == "gene expression RNAseq", Label == "IlluminaHiSeq")
str(ge)
```

Or I just copy `https://tcga.xenahubs.net` and `TCGA.LUAD.sampleMap/HiSeqV2`.

## Get Your Gene Values

Once you got dataset information, you can get a specific gene expression (it also works for gene-level CNV, mutation, etc based on your dataset) by `fetch_dense_values`. Run `?fetch` in your R console to see more details.

For example, I will query the gene `TP53`.

```{r}
TP53 <- fetch_dense_values(host = ge$XenaHosts,       # You can also set "https://tcga.xenahubs.net"
                          dataset = ge$XenaDatasets,  # You can also set "TCGA.LUAD.sampleMap/HiSeqV2"
                          identifiers = "TP53",
                          use_probeMap = TRUE) %>% 
  .[1, ]
head(TP53)
```

Typically, the TCGA sample ID have 15 letters, and the 14-15th letters mark a sample type. When it `<10`, it is a tumor sample, otherwise it is a normal sample.

```{r}
table(as.integer(substr(names(TP53), 14, 15)))
```

Now you can start your analysis with this data.

## Other Things May Help

In addition to `fetch_*` functions, I generated many low-level API functions for UCSC Xena database, which described at <https://shixiangwang.github.io/home/en/tools/ucscxenatools-api/>. These functions can access different levels of data information in UCSC Xena. Some of them are combined to construct the core functionalities provided by `UCSCXenaTools` for now.

NOTE: not API functions work well, I haven't tested them all, they are all generated by dynamic code based on [XQuery](https://github.com/ropensci/UCSCXenaTools/tree/master/inst/queries).

An R Shiny package [UCSCXenaShiny](https://github.com/openbiox/UCSCXenaShiny/) provides a web-based platform to download datasets and analyze single genes. Besides, we have constructed some functions to get pan-cancer level single gene expression, CNV and mutation etc.

You can download recent development version in GitHub with:

```r
remotes::install_github("openbiox/XenaShiny")
```

After you load this package, you can use the following functions to get data easily.

```
get_ccle_cn_value: Fetch copy number value from CCLE dataset

get_ccle_gene_value: Fetch gene expression value from CCLE dataset

get_ccle_protein_value: Fetch gene protein expression value from CCLE dataset

get_ccle_mutation_status: Fetch gene mutation info from CCLE dataset

get_pancan_value: Fetch identifier value from pan-cancer dataset

get_pancan_gene_value: Fetch gene expression value from pan-cancer dataset

get_pancan_protein_value: Fetch protein expression value from pan-cancer dataset

get_pancan_mutation_status: Fetch mutation status value from pan-cancer dataset

get_pancan_cn_value: Fetch gene copy number value from pan-cancer dataset processed by GISTIC 2.0
```

Any questions can be posted online at <https://github.com/openbiox/UCSCXenaShiny/issues> or <https://github.com/ropensci/UCSCXenaTools/issues>.

## References

- Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627
- Wang, S.; Xiong, Y.; Gu, K.; Zhao, L.; Li, Y.; Zhao, F.; Li, X.; Liu, X. UCSCXenaShiny: An R Package for Exploring and Analyzing UCSC Xena Public Datasets in Web Browser. Preprints 2020, 2020070179 (doi: 10.20944/preprints202007.0179.v1).
